### PR TITLE
Extern Specs

### PR DESCRIPTION
**Pull request by xldenis**
Need to clean things up a lot, but lays the groundwork for extern specs.

### CODE REVIEW

File: extern_test.rs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+extern crate creusot_contracts;
+use creusot_contracts::*;
+
+#[creusot::no_translate]
+#[creusot::extern_spec]
```
Comment (jhjourdan) [inline extern_test.rs:5]: Would it be possible to make `extern_spec` imply `no_translate` automatically?

File: extern_test.rs [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+#[requires(true === true)]
+fn extern_xxxx<T>() -> Vec<T> {
```
Comment (jhjourdan) [inline extern_test.rs:7]: I don't understand the syntax here. Is it supposed to be the spec of function `xxxx`? Why does this function have a body?

File: extern_test.rs (hunk 1 above)
Comment (xldenis) [reply extern_test.rs:5]: Yes, one of the final steps will be to wrap this all up in a macro. What you're seeing here is the desugared syntax

File: extern_test.rs (hunk 2 above)
Comment (xldenis) [reply extern_test.rs:7]: `xxxx` represents a random uuid. We attach the spec to `extern_xxxx` and inside have a call to the function we really want to specify. 

This structure is partially caused by the fact we can only resolve names after/during type checking. To even identify `Vec::new` we must run the type checker.

**Pull request merged**